Update regex splitter for multiple carets

shamshirz opened the pull request:
Change quote to hon_quote to avoid keyword issue
Add test around multiple caret issue

shamshirz committed:
Update regex splitter for multiple carets

Change quote to hon_quote to avoid keyword issue
Add test around multiple caret issue

rjdellecese commented on lib/hon_graffiti_phoenix/parsers/markup_parser.ex:
```diff
@@ -28,9 +28,9 @@ defmodule HonGraffitiPhoenix.Parsers.MarkupParser do
-    quote
+  def parse(hon_quote) when hon_quote == nil, do: []
+  def parse(hon_quote) do
+    hon_quote
```
We'll also want to change all other references, including the model name... Don't need to do right now, I suppose.

rjdellecese replied:
https://trello.com/c/bAeXBais

shamshirz replied:
word, yeah

rjdellecese commented on test/lib/markdown_parser_test.exs:
```diff
@@ -49,4 +50,12 @@ defmodule HonGraffitiPhoenix.MarkupParserTest do
     assert hd(MarkupParser.parse("^*")).color == "white"
   end
 
+  test "it can handle multiple carets" do
```
I think a more descriptive test name would be better, e.g.
```Elixir
test "groups multiple sequential carets into one body, except the last one" do
```

Also, what happens when the final caret is followed by an invalid color code? Such as

```Elixir
"We shouldn't freak over ^^^^^j many carets"
# or
"We shouldn't freak over ^^^^^1e2 many carets"
# or
"We shouldn't freak over ^^^^^ many carets"
```

shamshirz replied:
Updated, incoming

rjdellecese reviewed (changes requested):
I think we should make these tests follow the four-phase testing pattern (references [here](https://robots.thoughtbot.com/four-phase-test) and [here](http://xunitpatterns.com/Four%20Phase%20Test.html)). It would be cool to have all of the tests in this file follow this pattern, but if you only want to change the one you added here I'm ok with that too.

Also, as I mentioned, I think it's better to include the contents of the quotes you're testing in the context of each test, unless you're sharing it between different tests.

shamshirz committed:
Update tests to 4phase

shamshirz committed:
rename markdown parser file to markup

samgqroberts commented on test/lib/markup_parser_test.exs:
```diff
@@ -0,0 +1,80 @@
+  end
+
+  test "it returns the original string if there is no style" do
+    no_markup = "thisIsSoFakn' BORING"
```
can u mark this test nsfw pls

rjdellecese commented on test/lib/markup_parser_test.exs:
```diff
@@ -0,0 +1,80 @@
+    code_color = "^rwords"
+
+    rgb_parsed = MarkupParser.parse(rgb_color)
+    code_parsed = MarkupParser.parse(code_color)
```
I think these should be two different tests--one for parsing markup codes and one for parsing digits.

shamshirz replied:
Gotcha, yes, duh, much nicer that way.

rjdellecese commented on test/lib/markup_parser_test.exs:
```diff
@@ -0,0 +1,80 @@
+    invalid_multiple_carets = "Don't freak^^^^^1too many carets"
+
+    valid_parsed = MarkupParser.parse(valid_multiple_carets)
+    invalid_parsed = MarkupParser.parse(invalid_multiple_carets)
```
Likewise here--since there are two conditions being tested (valid case and invalid case) I think these should be two different tests. You could use a [`describe` block](https://hexdocs.pm/ex_unit/ExUnit.Case.html#describe/2) to distinguish between the two different contexts.

rjdellecese reviewed (changes requested):
Two more little things and then I think this is gtg 👍  I should have brought up these testing things during the first PR, my bad.

shamshirz committed:
Update / simplify markup tests

Add describe blocks for common tests, split larger tests into multiple
smaller tests

rjdellecese reviewed (approved):
Thanks for the bug-fix and test-tidying @Miguel4president 👍
